src/components/redux/actions/index.jsx

```diff
@@ -1,8 +1,89 @@
-import { FIRST_TYPE } from "./actionsTypes";
+import {
+  ADD_TO_CART,
+  DELETE_FROM_CART,
+  SUM_TOTAL_PRICE,
+  ADD_PRICE_OF_ONE_ITEM,
+  SUBTRACT_PRICE_OF_ONE_ITEM,
+  INCREASE_QUANTITY,
+  DECREASE_QUANTITY,
+} from "./actionsTypes";
 
-export const currentBoardIDAction = (data) => {
+// Adding items to the shopping cart with the selected quantity
+export const addProductToCart = (data) => {
   return {
-    type: FIRST_TYPE,
+    type: ADD_TO_CART,
     payload: data,
   };
 };
+
+// This function subtract one from the quantity which the user added to the shopping cart.
+const handleQuantitySubtraction = (data) => {
+  const copyOfData = data.slice();
+  for (let i = 0; i < copyOfData.length; i++) {
+    const currentItemId = data[i].id;
+    if (copyOfData[i].id === currentItemId) {
+      copyOfData[i].quantity--;
+    }
+  }
+  return copyOfData;
+};
+
+// This function add one to the quantity which the user added to the shopping cart.
+const handleQuantityAddition = (data) => {
+  const copyOfData = data.slice();
+  for (let i = 0; i < copyOfData.length; i++) {
+    const currentItemId = data[i].id;
+    if (copyOfData[i].id === currentItemId) {
+      copyOfData[i].quantity++;
+    }
+  }
+  return copyOfData;
+};
+
+// This action for handleQuantityAddition function
+export const increaseProductQuantity = (data) => {
+  return {
+    type: INCREASE_QUANTITY,
+    payload: () => handleQuantityAddition(data),
+  };
+};
+
+// This action for handleQuantitySubtraction function
+export const decreaseProductQuantity = (data) => {
```

Comment:
the action and the action type should not be different named. decreaseProductQuantity but handleQuantityProductSubtraction, it should just be handleX

Reply:
please fix in another places too